src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -208,6 +208,45 @@ pub trait TimestampProvider {
 
     async fn oracle_read_ts(&self, timeline: &Timeline) -> Option<Timestamp>;
 
+    fn get_timeline(timeline_context: &TimelineContext) -> Option<Timeline> {
+        let timeline = match timeline_context {
+            TimelineContext::TimelineDependent(timeline) => Some(timeline.clone()),
+            // We default to the `Timeline::EpochMilliseconds` timeline if one doesn't exist.
+            TimelineContext::TimestampDependent => Some(Timeline::EpochMilliseconds),
+            TimelineContext::TimestampIndependent => None,
+        };
+
+        timeline
+    }
```

Comment:
Almost, but I think in my case I'll need `Timeline::EpochMilliseconds` even for `TimelineContext::TimestampIndependent`, because I'd like to handle `mz_now()` in the `REFRESH AT` even if the query is a constant (which is the case where `validate_timeline_context` returns `TimelineContext::TimestampIndependent`).